addons/sale_stock/models/sale_order.py

```diff
@@ -289,7 +289,7 @@ def _get_action_view_picking(self, pickings):
             picking_id = picking_id[0]
         else:
             picking_id = pickings[0]
-        action['context'] = dict(default_partner_id=self.partner_id.id, default_picking_type_id=picking_id.picking_type_id.id, default_origin=self.name, default_reference_ids=self.stock_reference_ids[:-1].id)
+        action['context'] = dict(default_partner_id=self.partner_id.id, default_picking_type_id=picking_id.picking_type_id.id)
```

Comment:
```suggestion
action['context'] = dict(
    default_partner_id=self.partner_id.id,
    default_picking_type_id=picking_id.picking_type_id.id,
)
if len(self) == 1:
    action['context'].update(
        default_origin=self.name,
        default_reference_ids=self.stock_reference_ids[:-1].id
    )
```

I didn't check if what I wrote for the `default_reference_ids` makes sense

Reply:
Hello @ticodoo,

This still gives an error. The issue is that if a **SO has multiple linked MOs**, then using **self.stock_reference_ids[:-1].id** will raise a traceback.

For example, if **stock_reference_ids contains IDs [1, 2, 3, 4], then self.stock_reference_ids[:-1] returns [1, 2, 3], and accessing .id** on that gives a singleton error.

The purpose here is to remove the reference; you can refer to Dhha’s comment(https://github.com/odoo/odoo/pull/230270#discussion_r2410379542) for more context.